content/automate/ManageIQ/Cloud/VM/Provisioning/Placement.class/__methods__/best_fit_azure.rb

```diff
@@ -1,37 +1,64 @@
 ###################################################################
-#
-# Description: Select the cloud network and availability zone for Azure
-#
+#                                                                 #
+# Description: Select the cloud network, availability zone        #
+#              and resource group for Azure                       #
+#                                                                 #
 ###################################################################
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Provisioning
+          module Placement
+            class BestFitAzure
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-$evm.log("info", "Using Auto Placement for Azure Cloud Provider")
-prov  = $evm.root["miq_provision"]
-image = prov.vm_template
-raise "Image not specified" if image.nil?
+              def main
+                @handle.log("info", "Using Auto Placement for Azure Cloud Provider")
+                @prov = @handle.root["miq_provision"]
+                raise "Image not specified" if @prov.try(:vm_template).nil?
 
-if prov.get_option(:cloud_network).nil?
-  cloud_network = prov.eligible_cloud_networks.first
+                @prov.get_option(:cloud_network)  || default_cloud_network
+                @prov.get_option(:cloud_subnet)   || default_cloud_subnet
+                @prov.get_option(:resource_group) || default_resource_group
+              end
 
-  if cloud_network
-    prov.set_cloud_network(cloud_network)
-    $evm.log("info", "Selected Cloud Network: #{cloud_network.name}")
-  end
-end
+              private
```

Comment:
@pkomanek 
can we check if prov is present and raise an error, also prov could just be a method

something similar to
https://github.com/ManageIQ/manageiq-content/blob/09ad2ef2f24cbef3463f86fb460573149b98a6c3/content/automate/ManageIQ/Infrastructure/VM/Provisioning/Placement.class/__methods__/vmware_best_fit_least_utilized.rb#L22
```
def prov
   @prov ||= @handle.root["miq_provision"].tap do |req|
     raise "miq_provision not provided"
  end
end
```